repos/system_upgrade/el7toel8/libraries/isccfg.py

```diff
@@ -0,0 +1,959 @@
+#!/usr/bin/env python
+#
+# Simplified parsing of bind configuration, with include support and nested sections.
+
+import re
+import string
+
+
+class ConfigParseError(Exception):
+    """ Generic error when parsing config file """
+
+    def __init__(self, message, error=None):
+        super(ConfigParseError, self).__init__(message)
+        self.error = error
+    pass
+
+
+class ConfigFile(object):
+    """ Representation of single configuration file and its contents """
+    def __init__(self, path):
+        """
+        Load config file contents from path
+
+        :param path: Path to file
+        """
+        self.path = path
+        self.load(path)
+        self.status = None
+
+    def __str__(self):
+        return self.buffer
+
+    def __repr__(self):
+        return 'ConfigFile {0} ({1})'.format(
+                self.path, self.buffer)
+
+    def load(self, path):
+        with open(path, 'r') as f:
+            self.buffer = self.original = f.read()
+            f.close()
+
+    def is_modified(self):
+        return self.original == self.buffer
+
+    def root_section(self):
+        return ConfigSection(self, None, 0, len(self.buffer))
+
+
+class MockConfig(ConfigFile):
+    """ Configuration file with contens defined on constructor.
+
+        Intended for testing the library.
+    """
+    DEFAULT_PATH = '/etc/named/mock.conf'
+
+    def __init__(self, contents, path=DEFAULT_PATH):
+        self.original = contents
+        super(MockConfig, self).__init__(path)
+
+    def load(self, path):
+        self.buffer = self.original
+
+
+class ConfigSection(object):
+    """ Representation of section or key inside single configuration file.
+
+    Section means statement, block, quoted string or any similar. """
+
+    TYPE_BARE = 1
+    TYPE_QSTRING = 2
+    TYPE_BLOCK = 3
+    TYPE_IGNORED = 4  # comments and whitespaces
+
+    def __init__(self, config, name=None, start=None, end=None, kind=None, parser=None):
+        """
+        :param config: config file inside which is this section
+        :type config: ConfigFile
+        :param kind: type of this section
+        """
+        self.config = config
+        self.name = name
+        self.start = start
+        self.end = end
+        self.ctext = self.original_value()   # a copy for modification
+        self.parser = parser
+        if kind is None:
```

Comment:
maybe `if not kind`?

Reply:
Since kind may contain numeric constant, I would keep None as special value to notify. Value 0 should not pass this test, even it is not used.